xbmc/Application.cpp

```diff
@@ -3733,20 +3757,37 @@ void CApplication::CheckOSScreenSaverInhibitionSetting()
 
 void CApplication::CheckScreenSaverAndDPMS()
 {
-  bool maybeScreensaver =
-      !m_dpmsIsActive && !m_screensaverActive
-      && !m_ServiceManager->GetSettings().GetString(CSettings::SETTING_SCREENSAVER_MODE).empty();
-  bool maybeDPMS =
-      !m_dpmsIsActive && m_dpms->IsSupported()
-      && m_ServiceManager->GetSettings().GetInt(CSettings::SETTING_POWERMANAGEMENT_DISPLAYSOFF) > 0;
+  bool maybeScreensaver = true;
+  if (m_dpmsIsActive)
+    maybeScreensaver = false;
+  else if (m_screensaverActive)
+    maybeScreensaver = false;
+  else if (m_ServiceManager->GetSettings().GetString(CSettings::SETTING_SCREENSAVER_MODE).empty())
+    maybeScreensaver = false;
+
+  bool maybeDPMS = true;
+  if (m_dpmsIsActive)
+    maybeDPMS = false;
+  else if (!m_dpms->IsSupported())
+    maybeDPMS = false;
+  else if (m_ServiceManager->GetSettings().GetInt(CSettings::SETTING_POWERMANAGEMENT_DISPLAYSOFF) <= 0)
+    maybeDPMS = false;
```

Comment:
I do not see how this change makes the code easier to read. I think the contrary is the case because it makes the code less concise. You can read the original statement like a "sentence" and understand what it means. The variables and assignments and additional if statements just add visual clutter. Just my opinion of course.

I generally agree that very large if statements and ternaries can use some splitting/cleaning up and there are good examples in this PR, but I don't think we should start replacing every if condition that has two `&&` with flags.
Maybe we should discuss this on the forums and reach some kind of consensus?

Reply:
> Maybe we should discuss this on the forums and reach some kind of consensus?

to be fair. this PR was open for quite a while and everybody could comment. imo github is better suited for discussing code than the forum is. not sure if we can find a general rule that fits all different cases of how to order expressions. 

btw. having multiple function calls on a single statement makes debugging harder because you can't set breakpoints.

Reply:
Recognize that we're working within a framework of many different values. Conciseness is one value. Readability is another. In this case, my primary goal was scalability. These commits originally served to make the code more scalable for a subsequent fix (which I later abandoned).

Discussion of values is always a healthy thing. However, you're basically asking for consensus on the Trolley Problem. Each developer needs to weigh these values against each other and make their own decision. In my case, I chose to honor scalability over other values. This doesn't mean I was right, just that I chose a particular value to make a decision by.